paginator/paginator_paginate_test.go

```diff
@@ -669,6 +674,134 @@ func (s *paginatorSuite) TestPaginateCustomTypeString() {
 	s.assertIDs(p1, 9, 8, 7)
 }
 
+type NullString sql.NullString
+
+func (ns NullString) MarshalJSON() ([]byte, error) {
+	if !ns.Valid {
+		return []byte("null"), nil
+	}
+	return json.Marshal(ns.String)
+}
+
+func (ns *NullString) UnmarshalJSON(b []byte) error {
+	isNull := bytes.Equal(b, []byte("null"))
+	ns.Valid = !isNull
+
+	if isNull {
+		ns.String = "null"
+		return nil
+	}
+	return json.Unmarshal(b, &ns.String)
+}
+
+func (ns NullString) Value() (driver.Value, error) {
+	if !ns.Valid {
+		return nil, nil
+	}
+	return ns.String, nil
+}
+
+func (ns *NullString) Scan(value interface{}) error {
+	if value == nil {
+		ns.String = ""
+		ns.Valid = false
+	} else if strValue, ok := value.(string); ok {
+		ns.Valid = true
+		ns.String = strValue
+	} else {
+		return errors.New("unsupported type")
+	}
+	return nil
```

Comment:
```suggestion
	}
        return errors.New("unsupported type")
```

Reply:
I don't think this suggestion works here - it would return in any case.

Reply:
The else branch already covers all cases other than `value == nil` and `value.(string)`. So the `return nil` would never reach.

Reply:
Yes, but the `value == nil` and `value.(string)` branches don't have return statements, so we need the `return nil` for those cases somewhere. Instead of having it at the end, we could add `return nil` statements to inside those branches if you prefer that style - then we can have `return error` at the end as you suggested.

Reply:
Oh, yes. You are right. I miss to see that those two branches have no return 😢 Sorry about that. 

I think the PR is ready to go 💪 I will update the README and release a new version.

Reply:
Great, thank you :pray: :slightly_smiling_face: